### DISCUSSION TITLE
Documentation: Arrays in Dart/Flutter

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#6870] [General]
Hello, wondering if you can update the docs/guides for using Arrays in Flutter? Also, does Supabase support multidimensional arrays? If so, and example appreciated.

https://supabase.com/docs/guides/database/arrays

Thanks

**Comment by steve-chavez** [comment]
Hey there,

Yes, multidimensional arrays are supported. Here are [docs using the REST API](https://postgrest.org/en/latest/how-tos/working-with-postgresql-data-types.html#multidimensional-arrays). The patch example would have the same payload for supabase-js, query should have the same shape as well.

@dshukertjr Might know how to do this in flutter.

**Reply by dshukertjr** [reply]
Thanks @steve-chavez for tagging me! 

@jtkeyva Thanks for the pointing out the missing docs on array support! I will add Flutter/Dart version. I might also add the docs for multi dimensional arrays as well. Just out of curiosity, what kind of information are you storing in a multi dimensional array here?

**Comment by jtkeyva** [comment]
thanks all! @dshukertjr i'm dabbling with the idea of using it to store a list of tags with custom colors, custom stickers or emojis per room in a chat app. rather than have to do a join or another db call, its built right into the table. considered jsonb but read arrays are quicker and better when searching or filtering.
thoughts/feedback appreciated

**Reply by dshukertjr** [reply]
@jtkeyva I think it's okay to store list of tags as long as the number of tags in each row is within a reasonable number. You could use constraints to limit the maximum number of tags created in a row. 
https://stackoverflow.com/questions/39502496/how-to-limit-the-length-of-an-array-in-postgresql

Note that the query should not slow down as long as you set proper index even if you store these tag data in a separate table. Querying foreign table with Supabase is pretty straight forward, and some people might prefer having separate tables, just because they can tighten their data up more this way! It will come down to personal preference at the end though!
https://supabase.com/docs/reference/dart/select#query-foreign-tables

**Comment by jtkeyva** [comment]
As always, thanks for the thorough response! @dshukertjr 

ps: just noticed you put the same link twice. was the bottom link supposed to relate to querying a foreign table in supabase?

**Reply by dshukertjr** [reply]
Thanks @jtkeyva! I have fixed the link!
